src/components/Banners/SlackNotificationBanner.tsx

```diff
@@ -0,0 +1,215 @@
+  const settingsSlackNotificationPatchFail =
+    userSettingsData?.userSettings?.notifications?.patchFirstFailure;
+
+  console.log(
+    "settingsSlackNotificationPatchFinish :>> ",
+    settingsSlackNotificationPatchFinish
+  );
+  console.log(
+    "settingsSlackNotificationPatchFail :>> ",
+    settingsSlackNotificationPatchFail
+  );
+
+  // UPDATE USER SETTINGS MUTATION
+  const [
+    updateUserSettings,
+    { loading: loadingUpdateUserSettings },
+  ] = useMutation<
+    UpdateUserSettingsMutation,
+    UpdateUserSettingsMutationVariables
+  >(UPDATE_USER_SETTINGS, {
+    onCompleted: () => {
+      hideBanner();
+      dispatchBanner.successBanner(
+        "You will now receive Slack notifications when your patches fail or succeed"
+      );
+    },
+    onError: (err) => {
+      dispatchBanner.errorBanner(
+        `Error while saving settings: '${err.message}'`
+      );
+    },
+    refetchQueries: ["GetUserSettings"],
+  });
+
+  const saveNotificationSettings = () => {
+    if (!slackUsername) {
+      return;
+    }
+    updateUserSettings({
+      variables: {
+        userSettings: {
+          slackUsername,
+          notifications: {
+            patchFinish: "slack",
+            patchFirstFailure: "slack",
+          },
+        },
+      },
+    });
+  };
+
+  const cookieExists = cookie === doNotShowslackNotificationBannerCookie;
+
+  const showSlackBanner =
+    !cookieExists &&
+    settingsSlackNotificationPatchFail === "" &&
+    settingsSlackNotificationPatchFinish === "";
+
+  return (
+    <StyledPageWrapper>
+      <Banners
+        banners={bannersState}
+        removeBanner={dispatchBanner.removeBanner}
+      />
+      {showSlackBanner && (
+        <StyledBanner
+          bannerTheme="information"
+          data-cy="slack-notification-banner"
+        >
+          <Wrapper>
+            <StyledIcon glyph="InfoWithCircle" fill={uiColors.blue.base} />
+            <StyledBody>
+              You can receive a Slack notification when your patch is ready.
+            </StyledBody>
+            <Popconfirm
+              title={
+                <>
+                  <SlackUsername weight="medium">Slack Username</SlackUsername>
+                  <Input
+                    data-cy="slack-username-input"
+                    value={slackUsername}
+                    onChange={(e) => setSlackUsername(e.target.value)}
+                    onPressEnter={() => saveNotificationSettings()}
+                  />
+                </>
+              }
+              onConfirm={() => saveNotificationSettings()}
+              onCancel={() => setSlackUsername("")}
+              okText="Save"
+              cancelText="Cancel"
+              okButtonProps={{ loading: loadingUpdateUserSettings }}
+              cancelButtonProps={{ disabled: loadingUpdateUserSettings }}
+              icon={null}
+            >
+              <div data-cy="subscribe-to-notifications">
```

Comment:
This div is not needed.

Reply:
nvm it is needed. But its because emotion cant properly use the ref provided by the popconfirm. Thats strange.